src/domain/template/member-guidelines-template/dto/member.guidelines.template.dto.create.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  MID_TEXT_LENGTH,
+  VERY_LONG_TEXT_LENGTH,
+} from '@common/constants/entity.field.length.constants';
+import { Markdown } from '@domain/common/scalars/scalar.markdown';
+import { CreateTemplateBaseInput } from '@domain/template/template-base/dto';
+import { Field, InputType } from '@nestjs/graphql';
+import { MaxLength } from 'class-validator';
+
+@InputType()
+export class CreateMemberGuidelinesTemplateInput extends CreateTemplateBaseInput {
+  @Field({
+    nullable: false,
+    description: 'The type of Member Guidelines created from this Template.',
+  })
+  @MaxLength(MID_TEXT_LENGTH)
+  type!: string;
+
+  @Field(() => Markdown, {
```

Comment:
@techsmyth  can you please elaborate? Because if we have `CommunityGuidelines` reference, we change the reference, we change the template. This will create kinda dynamic template that changes all the time when the reference changes, if I understand correctly.

Reply:
@valentinyanakiev the idea is to use the full CommunityGuidelines entity to store the template information - but that will be cloned / copied when using the template. It is not shared across usages. The idea is to get towards having an approach where each template type simply wraps around a domain model entity - not duplicating the setup with a selection of the fields that would ultimately be used. Makes both maintenance easier + I think later we can actually move to a single template type that has one of a number of types (like the CalloutContribution entity)

Reply:
The current impl copies the content of the entity at that moment in time and creates a public version of it (template).
If i am understanding it correctly, you suggest to maintain a direct relation to the actual entity which is fundamentally different. Please contact me to discuss.

Reply:
It should clone/copy the entity wrapped by the template at the moment in time when the template is used to instantiate a new entity. The new entity should **not** maintain a direct relation.

Reply:
What do you mean by wrap. The implementation will follow the guideline entity, so the fields from the profile will be used copied to the fields in the profile of the template.
Be aware that you are commenting a PR not ready for a review and this is not the final implementation.

Reply:
### The final implementation will be:
When a template is created from input that will create a new entity in it's table and the template will mantain a 1:1 relation to that entry.
When a template is created from entity, a new entity from the same type will be duplicated and a 1:1 relation will be maintain from the template.

**This approach will take care of domain updates to the entity and will reduce the custom code and fields that have to be mainated with every new template type.**